tests/checkers/unittest_base.py

```diff
@@ -299,3 +299,81 @@ def test_no_six(self):
             has_six = False
 
         self.assertFalse(has_six, "pylint must be able to run without six")
+
```

Comment:
There's no need to duplicate the tests here, functional tests are better and enough. You can move all the example that are different to functional.